src/types/middleware.ts

```diff
@@ -108,11 +92,9 @@ export const contextBuiltinKeys: string[] = [
   'botUserId',
   'teamId',
   'enterpriseId',
-  'functionBotAccessToken',
-  'functionExecutionId',
-  'functionInputs',
   'retryNum',
   'retryReason',
+  ...Array<keyof CustomFunctionContext>(),
```

Comment:
I'm partial to both approaches! The magic is nice to avoid forgetting to add it to context and needing to hunt for these fields, but magic is a dangerous art 😳